week1-project/ecdsa-node/server/index.js

```diff
@@ -1,31 +1,90 @@
-const express = require("express");
+/* eslint-disable no-console */
+const express = require('express');
+
 const app = express();
-const cors = require("cors");
+const cors = require('cors');
+const secp = require('ethereum-cryptography/secp256k1');
+const { toHex } = require('ethereum-cryptography/utils');
+
 const port = 3042;
+const { utf8ToBytes } = require('ethereum-cryptography/utils');
 
 app.use(cors());
 app.use(express.json());
 
-const balances = {
-  "0x1": 100,
-  "0x2": 50,
-  "0x3": 75,
-};
+const actors = ['zahra', 'kate', 'constance', 'martina'];
+const actorKeyPairs = {};
+const balances = {};
+
+function generateKeyPairs() {
+  actors.forEach((item) => {
+    const privateKey = toHex(secp.utils.randomPrivateKey());
+    actorKeyPairs[`${item}`] = {
+      private_key: privateKey,
+      public_key: toHex(secp.getPublicKey(privateKey)),
+    };
+  });
+  return actorKeyPairs;
+}
+generateKeyPairs();
+// eslint-disable-next-line no-console
+console.log(JSON.stringify(actorKeyPairs));
+
+actors.forEach((actor, index) => {
+  balances[actorKeyPairs[actor].public_key] = Math.trunc((100 / (index + 1)));
+});
+
+async function signMessage(msg, privateKey) {
+  return secp.sign(msg, privateKey, { recovered: true });
+}
+
+async function recoverKey(message, signature, recoveryBit) {
+  return secp.recoverPublicKey(message, signature, recoveryBit);
+}
+
+function setInitialBalance(address) {
+  if (!balances[address]) {
+    balances[address] = 0;
+  }
+  return balances[address];
+}
```

Comment:
Does this end up clobbering the value set above on line 34?

Reply:
No because this is only called if there isn't an address defined. If there is one defined it just returns the already defined balance.